### PR TITLE
Bedrock improvements

### PR DESCRIPTION
**Pull request by tinydylan**
Related case: TINY-6136

Changes:
* Added `promiseTest` to test async functions (functions returning promises)
* Changed `asynctest` to `asyncTest`, following camelCase conventions. Old spelling remains for compatibility.
* Changed `asyncTest` so that it fails (instead of timing out) when an exception is thrown during execution of the test method.
* Remove the sample "suite" tests. The failing ones were causing timeouts.

Will release as new minor version: 9.6.0

### CODE REVIEW

File: modules/client/src/main/ts/api/UnitTest.ts [hunk 1]
```diff
@@ -115,3 +125,9 @@ export const test = (name: string, test: () => void) => {
     }
   });
 };
+
+/** Tests an async function (function that returns a Promise). */
+export const promiseTest = (name: string, test: () => Promise<void>) =>
+  asyncTest(name, (success, failure) => {
+    test().then(success).catch(failure);
```
Comment (TheSpyder) [inline modules/client/src/main/ts/api/UnitTest.ts:132]: We've had this in the past, I'm not sure why it was removed. Polyfill required on IE11, anyway.

**Comment by lnewson** [comment]
Removed my changes requested, as per the discussion on slack. Not approving though, because of the issue Andy mentioned.

File: modules/client/src/main/ts/api/UnitTest.ts (hunk 1 above)
Comment (tinydylan) [reply modules/client/src/main/ts/api/UnitTest.ts:132]: As discussed out-of-band, we'll leave it up to the caller to polyfill Promises.

File: modules/sample/src/test/ts/client/AsyncFailTest.ts [hunk 2]
```diff
@@ -1,10 +1,26 @@
 import { UnitTest } from '@ephox/bedrock-client'
 
-UnitTest.asynctest('AsyncFail Test', (success, failure) => {
+UnitTest.asyncTest('AsyncFail Test 1', (success, failure) => {
```
Comment (lnewson) [inline modules/sample/src/test/ts/client/AsyncFailTest.ts:3]: This test still needs to import a promise polyfill from somewhere.

- [ ] Add Promise polyfill in this test

File: modules/sample/src/test/ts/client/AsyncPassTest.ts [hunk 3]
```diff
@@ -1,10 +1,24 @@
 import { UnitTest } from '@ephox/bedrock-client'
 
-UnitTest.asynctest('AsyncPass Test', (success, failure) => {
-
+UnitTest.asyncTest('AsyncPass Test 1', (success, failure) => {
   new Promise(function (resolve, reject) {
```
Comment (lnewson) [inline modules/sample/src/test/ts/client/AsyncPassTest.ts:4]: Interesting that this didn't have a poyfill previously... it really should have.

Comment (tinydylan) [reply modules/sample/src/test/ts/client/AsyncPassTest.ts:4]: Yeah, looks like CI is only running these tests in phantomjs.

Comment (tinydylan) [reply modules/sample/src/test/ts/client/AsyncPassTest.ts:4]: I've logged internal ticket TINY-6159 to add tests cross-browser.

Comment (tinydylan) [reply modules/sample/src/test/ts/client/AsyncPassTest.ts:4]: When I add `@ephox/wrap-promise-polyfill` and try to import it, I get:

```
TS2529: Duplicate identifier 'Promise'. Compiler reserves name 'Promise' in top level scope of a module containing async functions.
```

Comment (tinydylan) [reply modules/sample/src/test/ts/client/AsyncPassTest.ts:4]: I'm going to add the polyfill fix to TINY-6159. Sound good?

File: modules/client/src/main/ts/api/UnitTest.ts (hunk 1 above)
Comment (TheSpyder) [reply modules/client/src/main/ts/api/UnitTest.ts:132]: That's fine for tests _using_ bedrock, but not bedrock's tests.

File: modules/sample/src/test/ts/client/AsyncPassTest.ts (hunk 3 above)
Comment (TheSpyder) [reply modules/sample/src/test/ts/client/AsyncPassTest.ts:4]: That's a typescript error? weird, we don't have that issue in other tests, oh well.

Comment (tinydylan) [reply modules/sample/src/test/ts/client/AsyncPassTest.ts:5]: As discussed: polyfill has been added. It doesn't work with `async..await` - seems like TS requires a global `Promise` for that. For now, I've removed all `async..await`.

**Pull request merged**